inst/include/dust/interface.hpp

```diff
@@ -40,6 +44,44 @@ cpp11::list dust_alloc(cpp11::list r_data, int step,
   return cpp11::writable::list({ptr, info});
 }
 
+// think about rng seeding here carefully; should accept either a raw
+// vector or an integer I think.
+template <typename T>
+cpp11::writable::doubles dust_simulate(cpp11::sexp r_steps,
+                                       cpp11::list r_data,
+                                       cpp11::doubles_matrix r_state,
+                                       cpp11::sexp r_index,
+                                       const size_t n_threads,
+                                       const size_t seed) {
+  typedef typename T::real_t real_t;
+  std::vector<size_t> steps = validate_size(r_steps, "steps");
+  // TODO: broken for floats
```

Comment:
because cpp11 is `doubles_matrix` (and R doesn't have floats)? Could we cast everything at the right point?
One thing I've never worked out in C++ is using conditionals based on a template type (maybe you aren't supposed to be able to do that for a reason)

Reply:
ha, it's no longer broken for floats as I fixed that